src/Approximations/template_directions.jl

```diff
@@ -330,3 +337,127 @@ end
 
 end # @eval
 end # if
+
+# ==================================================
+# Spherical directions
+# ==================================================
+
+@inline _isapprox_included(d, S) = any(isapproxzero.([norm(d - si, Inf) for si in S])) 
```

Comment:
I have the impression that this first creates the whole vector and only then starts evaluating. Is my impression wrong?

Reply:
That's correct. I've revised the implementation, please reopen if you have further comments.